pkg/adaptor/adaptor.go

```diff
@@ -24,15 +23,21 @@ func (a ErrAdaptor) Error() string {
 // Listen() listens on a pipe, processes data, and then emits it.
 // Stop() shuts down the adaptor
 type StopStartListener interface {
+	SampleConfig() string
+	Description() string
+	Connect() error
```

Comment:
it seems as though this one interface is trying to do too much, 
you're combining a readwritecloser with a documentation and connect fn's
it may be more appropriate to split this up into a few different interfaces, 
i.e. a readwritecloser, a connector, and a documentor